src/SoftDeleteable/Query/TreeWalker/SoftDeleteableWalker.php

```diff
@@ -20,14 +24,50 @@
  */
 class SoftDeleteableWalker extends SqlWalker
 {
+    /**
+     * @var Connection
+     *
+     * @deprecated to be removed in 4.0, use the getConnection method instead
+     */
     protected $conn;
```

Comment:
The connection is available through a public getter on the parent class, no need to have a duplicate reference here.